force-app/main/default/classes/OpportunityHelper.cls

```diff
@@ -0,0 +1,146 @@
+        Map<Id, Account> accounts = new Map<Id, Account>([SELECT Id, Industry FROM Account WHERE Id IN (SELECT AccountId FROM Opportunity WHERE Id IN :oldOpps)]);
+        for(Opportunity opp : oldOpps){
+            if (opp.isClosed) {
+                opp.addError('Cannot delete closed opportunity');
+            } else if (opp.StageName == 'Closed Won'){
+                if(accounts.get(opp.AccountId).Industry == 'Banking'){
+                    opp.addError('Cannot delete closed opportunity');
+                }    
+            }    
+        }
+    }
+
+    public static void setType (List<Opportunity> newOpps) {
+        for (Opportunity opp : newOpps) {
+            if (opp.Type == null){
+                opp.Type = 'New Customer';
+            }    
+        }
+    }
+
+    public static void setPrimaryContact (List<Opportunity> newOpps) {
+        //Get contacts related to the opportunity account
+        Set<Id> accountIds = new Set<Id>();
+        for(Opportunity opp : newOpps){
+            accountIds.add(opp.AccountId);
+        }
+        
+        Map<Id, Contact> contacts = new Map<Id, Contact>([SELECT Id, FirstName, AccountId FROM Contact WHERE AccountId IN :accountIds AND Title = 'CEO' ORDER BY FirstName ASC]);
+        Map<Id, Contact> accountIdToContact = new Map<Id, Contact>();
+
+        for (Contact cont : contacts.values()) {
+            if (!accountIdToContact.containsKey(cont.AccountId)) {
+                accountIdToContact.put(cont.AccountId, cont);
+            }
+        }
+
+        for(Opportunity opp : newOpps){
+            if(opp.Primary_Contact__c == null){
+                if (accountIdToContact.containsKey(opp.AccountId)){
+                    opp.Primary_Contact__c = accountIdToContact.get(opp.AccountId).Id;
+                }
+            }
+        }
+    }
+
+    public static void insertTask (List<Opportunity> newOpps) {
+        List<Task> tasksToInsert = new List<Task>();
+            for (Opportunity opp : newOpps){
+                Task tsk = new Task();
+                tsk.Subject = 'Call Primary Contact';
+                tsk.WhatId = opp.Id;
+                tsk.WhoId = opp.Primary_Contact__c;
+                tsk.OwnerId = opp.OwnerId;
+                tsk.ActivityDate = Date.today().addDays(3); 
+                tasksToInsert.add(tsk);
+            }
+            insert tasksToInsert;
+    }
+
+    public static void updateDescriptionFromStage (List<Opportunity> newOppList, Map<Id, Opportunity> oldOppMap) {
+        List<Opportunity> oppsToUpdate = new List<Opportunity>();
+
+        for (Opportunity opp : newOppList){
+            Opportunity oldOpp = oldOppMap.get(opp.Id); 
+            if (opp.StageName != null && !opp.StageName.equals(oldOpp.StageName)){
+                Opportunity oppToUpdate = opp.clone(false, true);
+                oppToUpdate.Id = opp.Id;
+                oppToUpdate.Description += '\n Stage Change:' + opp.StageName + ':' + DateTime.now().format();
```

Comment:
What happens if the description value is blank/null initially? Try this out in the org.